docs/ref/signals.txt

```diff
@@ -195,6 +231,22 @@ Arguments sent with this signal:
 ``using``
     The database alias being used.
 
+Alternative to the pre_delete signal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Unless the model you are targetting is provided by a library and cannot be swapped, a
+more robust and maintainable solution would be to override the model's
+:meth:`~django.db.models.Model.delete` method::
+
+    class Order(models.Model):
+        ...
+        def delete(self, **kwargs):
```

Comment:
Unless we use `*args`, the function signature will be changed:

```suggestion
        def delete(self, *args, **kwargs):
```

Reply:
@adamchainz not sure I follow, should we add `*args` even though the `.delete()` method doesn't have any?

Reply:
The arguments to `delete()` can be passed *either* as positional or keywoard arguments. So *technically* to be a good overridder, you should allow both. This will also be compatible with any possible future changes to the signature. This applies to all the override examples in this PR.